### DISCUSSION TITLE
Datatable : FilteredValue always null

### DISCUSSION THREAD

**Discussion by kurtomerfaruk** [primefaces/community#2961] [PrimeFaces] [answered]
I want to get the money columns as subtotal in the datatable. I want to do this operation when filtered, so when I want to use the filteredValue operation, it always returns null. If I want to pull this operation lazyModel, it gives the result of the previous query.

`<p:ajax event="filter" listener="#{dtLazyView.onFilter}" update=":columnFooter,:growl"/>`

`<p:columnGroup id="columnFooter" type="footer">
                    <p:row>
                        <p:column colspan="4" footerText="Total :" styleClass="text-right font-bold"/>
                        <p:column id="subTotal" footerText="#{dtLazyView.subTotal}"
                                  class="font-bold">
                        </p:column>
                    </p:row>
</p:columnGroup>`

`    public void onFilter(AjaxBehaviorEvent event) {
        subTotal = BigDecimal.ZERO;
        for (History lazyItem : this.getLazyItems()) {
            subTotal= subTotal.add(lazyItem.getTotal());
        }
    }`

**Comment by melloware** [comment]
Please provide an executable example using the [PrimeFaces Test](https://github.com/primefaces/primefaces-test) project.  It is the only way developers can debug your problem to help.

**Comment by kurtomerfaruk** [comment]
You can reach from the link below
https://github.com/kurtomerfaruk/primefaces-test-master

**Reply by melloware** [reply]
your example doesn't even compile when i do `mvn clean jetty:run`

```
t primefaces-test: Compilation failure
[ERROR] /C:/temp/primefaces/discussion-2961/src/main/java/org/primefaces/test/LazyView.java:[65,31] cannot find symbol
[ERROR]   symbol:   class AjaxBehaviorEvent
[ERROR]   location: class org.primefaces.test.LazyView
```

**Reply by melloware** [reply, marked as answer]
@kurtomerfaruk i fixed your example:
[discussion-2961.zip](https://github.com/user-attachments/files/17620826/discussion-2961.zip)

I think your issue is that Ajax Filter is Before the filtering not AFTER.  The docs say https://primefaces.github.io/primefaces/15_0_0/#/components/datatable?id=ajax-behavior-events "On Filtering" which implies its before the filtering occurs.  You want it AFTER the filtering has occurred but i am pretty sure its before to give the user the opportunity to do something before filtering.

The AJAX filter event provides you the Filter Meta to manipulate.

```java
public class FilterEvent extends AbstractAjaxBehaviorEvent {

    private static final long serialVersionUID = 1L;

    private Map<String, FilterMeta> filterBy;

    public FilterEvent(UIComponent component, Behavior behavior, Map<String, FilterMeta> filterBy) {
        super(component, behavior);
        this.filterBy = filterBy;
    }

    public Map<String, FilterMeta> getFilterBy() {
        return filterBy;
    }
}
```

So you could in theory on that method simply call the same code as your lazy filter to produce a filtered list.
